docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
+
+**Implementation**
+
+When the user attempts to edit a profile, the ProfileSession, Ui, ProfileParser, Command, CommandLib, ProfileStorage, Profile and CommandResult classes will be accessed, and the following sequence of actions is called to prompt execution result to user:
+
+1. User executes `edit /w 60`
+    1. `ProfileSession` calls `Ui.getCommand()` to receive user input.
+    1. `ProfileSession` calls `ProfileParser.parseCommand()` to parse user input into a string array.
+1. Creating `ProfileEdit` object.
+    1. Based on the parsed input, `ProfileSession` calls `CommandLib` to return the correct Command Object `ProfileEdit`.
+1. Executing command.
+    1. `ProfileSession` calls `ProfileEdit.execute()` with the rest of parsed input.
+    1. `ProfileEdit` calls `ProfileStorage.loadData()` to load existing profile in the system. If there is no existing profile, `ProfileAdd` returns a failure result to `ProfileSession`. Otherwise, the process continues with step `iii`.
+    1. `ProfileEdit` calls `ProfileParser.extractCommandTagAndInfo()` to parse user input into specific tags and information.
+    1. Based on the parsed information from `ProfileParser.extractCommandTagAndInfo()`, `ProfileEdit` creates a new `Profile`.
+    1. `ProfileEdit` calls `Profile.equals()` to compare the edited and existing profile. If there are no changes, `ProfileEdit` returns a failure result to `ProfileSession`. Otherwise, the process continues with step `vi`.
+    1. `ProfileEdit` calls `ProfileStorage.saveData()` to save the edited `Profile` object.
+    1. `ProfileAdd` returns a successful result to `ProfileSession`.
+1. Prompting result to user.
+    1. `ProfileSession` calls `CommandResult.getFeedbackMessage()` to get the execution feedback message.
+    1. `ProfileSession` calls `Ui.showToUser()` to show result to the user.
+
+All descriptions, warnings and responses will be handled by `Ui` to ensure consistence across the app.
+
+The sequence diagram below summarizes how creating a new profile works:
+
+![Load Data Sequence Diagram](pictures/khoa/EditProfile.png)
+
+**Design considerations**
+
+Parsing of the user’s input command:
+
+- Alternative 1 (current choice): User’s command is split into size 2 array first containing command type and command arguments. Then arguments are split into command tag and information pairs.  
+
+|     |     |
+|-----|-----|
+|**Pros** | Command tags do not have to follow a fixed order.|
+|**Cons** | It takes multiple steps in parsing the command.|
+
+- Alternative 2: User’s command is divided by space
+                 
+|     |     |
+|-----|-----|
+|**Pros** | The parsing can be easily done by calling Java built-in function .split(). Supports multiple tags or no tags.|
+|**Cons** | Values for each variable cannot contain spaces which makes the application restrictive.|
+
+[&#8593; Return to Top](#developer-guide)
+
+#### 4.2.4. <a id="deleting-a-profile">Deleting a Profile</a>
```

Comment:
![Screenshot 2020-10-29 at 12 08 57 PM](https://user-images.githubusercontent.com/69417796/97526207-9dc2ec00-19e3-11eb-80e1-3e3b79e072ac.png)
Is there a lack of [else] condition?